### ISSUE TITLE
Component instances not isolating model correctly

### ISSUE THREAD

**Issue by rhasson** [emberjs/ember.js#11356]
My use case is as follows:  I allow my users to select from a number of widgets to be added to their workflow. They may add multiple widgets of the same kind but would have different content associated with them.

From a technical standpoint, I created a CollectionView (also tried this with ViewContainer) to keep the widgets together and give me a simple way to quickly add/remove widgets from the screen.  When the user selects a widget, I create a model for that widget and associated with a new instance of the widget's component (I also tried using views).  Then I add the newly created component to the CollectionView which renders it to the screen.  

My expectation is that each instance of the component I create will have its own copy of the model and will manipulate it independently.  However, if I create the same widget twice (assigning it two different instances of the model), they end up showing the same data.

In the code example below, click the button, type something into the input box that shows up.  Then click the button again.  The same text from the first input box shows on the second one.  Typing in either box update the text in both.

I've tested this on Ember v.1.12.0 - v.1.13.0

Here is a jsFiddle:  https://jsfiddle.net/azvccs3p/1/

Is this a bug or am I implementing something incorrectly?

Thanks,
Roy

<script type="text/x-handlebars" data-template-name="application">
	Hello:
	<button type="button" id="btnCreateIvr" {{action "createIvrAction"}}>Create</button>
	<hr>
	{{view App.containerView}}
</script>
<script type="text/x-handlebars" data-template-name="components/ivr-say">
	<div>This is an IVR-SAY component</div>
	{{log model}}
	{{input type="text" placeholder="Enter some text" value=text}}
</script>

<script type='application/javascript'>

	App = Ember.Application.create();
 
	App.ApplicationController = Ember.Controller.extend({
		actions: {
			createIvrAction: function() {
				var model, comp, view;
				var id = Date.now();

				model = this.store.createRecord('ivrsay', {
					id: id,
					index: id
				});

				comp = Ember.Component.extend({
					templateName: 'components/ivr-say',
					model: model,
					text: Ember.computed.alias('model.nouns.text'),
					didInsertElement: function() {
						console.log('inserted ', this);
					}
				});
				view = App.containerView.createChildView(comp);
				App.containerView.pushObject(view);
			}
		}
	});

	App.IvrSayComponent = Ember.Component.extend({
		templateName: 'components/ivr-say'
	});

	App.containerView = Ember.CollectionView.create({
		elementId: 'ivrcontainerview',
		classNames: ['col-md-12'],
		content: []
	});

	App.Ivrsay = DS.Model.extend({
		index: DS.attr('number'),
		verb: 'say',
		nouns: {
			text: null
		},
		verb_attributes: {
			voice: 'Woman',
			loop: 1,
			language: 'en'
		},
		params: {
			voice_options: ['Woman', 'Man']
		}
	});

**Comment by topaxi** [comment]
As rwjblue said in this issue https://github.com/emberjs/ember.js/issues/11354#issuecomment-109257917, you shouldn't set javascript objects (and arrays, which are objects) as properties.

**Comment by rhasson** [comment]
Thanks that solved my problem.

**Comment by rhasson** [comment]
Hi

I followed the direction on issue #11354 and made the corrections to my component.  However it's still broken.
When Iooking at the prototype chains I see that the DS.model I create for each of my components share the same prototype. Since I'm creating a new instance of the same DS.model for every one of my components the data I enter into one shows up in all the others. 
To test this out, I created an instance of the model but this time explicitly set the objects bound to the input box in the attributes object during model creation.  When I did that everything worked as expected and the model data was not shared across components.  With that said, I'm lead to believe there is an issue when how things are handled when models of the same type are created.

This JSBin shows the issue-
JSBin: http://jsbin.com/mavaxomofi/1/edit?html,js,output
To use it, just click the button next to the Hello, then type something in the box and click the button again. This will create another instance of the model/component but will show the text from the previous input box. I expect them to not share the data.

Thanks,
Roy
